packages/core/src/components/TextField/Mask.jsx

```diff
@@ -260,32 +277,28 @@ Mask.propTypes = {
 };
 
 /**
- * Remove mask characters from value
+ * Remove mask characters from value, or the same value back if invalid numeric string
  * @param {String} value
  * @param {String} mask
  * @returns {String}
  */
-export function unmask(value, mask) {
-  if (!value || typeof value !== 'string') return value;
-  const rawValue = value;
-  value = value.trim();
-
-  if (mask === 'currency') {
-    // Preserve only digits, decimal point, or negative symbol
-    const matches = value.match(/^-|[\d.]/g);
-    if (matches) {
-      value = matches.join('');
-    } else {
-      value = '';
+export function unmaskValue(value, mask) {
+  if (isValueMaskable(value, mask)) {
+    if (mask === 'currency') {
+      // Preserve only digits, decimal point, or negative symbol
+      const matches = value.match(/^-|[\d.]/g);
+      if (matches) {
+        value = matches.join('');
+      }
+    } else if (deliminatedMaskRegex[mask]) {
+      // Remove the deliminators and revert to single ungrouped string
+      value = toDigitsAndAsterisks(value);
     }
-  } else if (Object.keys(deliminatedMaskRegex).includes(mask)) {
-    // Remove the deliminators and revert to single ungrouped string
-    value = toDigitsAndAsterisks(value);
-  } else {
-    return rawValue;
   }
 
   return value;
 }
+// TODO: Remove deprecated naming
```

Comment:
I renamed from `unmask` to `unmaskValue` because [this line](https://github.com/CMSgov/design-system/pull/435/files#diff-bc403a3448d8591da046ffb97228eab1R147) suggests `unmaskValue` is the intended name. Also this makes it consistent with `maskValue`.